code/game/atoms_movable.dm

```diff
@@ -1101,9 +1101,13 @@
 			for(var/client/C in clients)
 				C.images -= hitmarker
 
-/atom/movable/proc/make_invisible(var/source_define, var/time, var/include_clothing)	//Makes things practically invisible, not actually invisible. Alpha is set to 1.
+//Makes things practically invisible, not actually invisible. Alpha is set to 1.
+/atom/movable/proc/make_invisible(var/source_define, var/time, var/include_clothing, var/alpha_value = 1, var/invisibility_value = 0)
 	return invisibility || alpha <= 1	//already invisible
```

Comment:
Why is this called "make_invisible" but it just returns a value?

Reply:
I have no idea, but I don't want to mess with it. If you can propose a change to this, I'll implement it

Reply:
Okay, I figured it out. There are some references to atom/moveable... Seems like it's part of a larger bug. I'lll fix it

Reply:
Digging a little deeper, it looks like all the child types override make_invisible() and call this to see if it's invisible already.  

This is a problem, as /atom/movable is the parent type of pretty much everything (except turfs and a few other things).  This means that you have to find each and every child of /atom/movable and make *absolutely certain* that this method (and the one below) are overridden properly or they'll bug out when this method is called.

What I would suggest is having a separate method called something like `is_invisible()` that performs this check separately.  That change would make this a bit more robust, it would make the child code clearer, and you can fill this instance of `make_invisible` with some quick and dirty fallback in case a tailored child method isn't available to make things invisible "properly".

Reply:
No more children. As god intended for this shitcode